### DISCUSSION TITLE
Runner Scale Set Controller Doesn't Seem to Pickup a new githubConfigUrl

### DISCUSSION THREAD

**Discussion by cameronrosier** [community/community#160470] [Actions]
Hey all, really sorry is this is the incorrect place to ask this - please lead me to the right place and I can move it.

I am trying to run the scale set controller along with the runners via helm on my k3s setup.  It was working well for a while, but I decided to change from a single github repository to a github organization, and when I did that, the controller still only tries to connect to the old github repository url.

I first deployed like this:
```bash
helm upgrade --install arc \
    --namespace "arc-systems" \
    --create-namespace \
    oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller
```

```
helm upgrade --install arc-runners \
    --namespace arc-runners \
    --create-namespace \
    --set githubConfigUrl="<url-to-github-repo>" \
    --set githubConfigSecret.github_token="<github PAT>" \
    oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set
```

This was running fine for the single github repository, but I wanted to move to an organization wide set of runners so I uninstalled the helm charts and tried to reinstall like:

```
helm upgrade --install arc \
    --namespace "arc-systems" \
    --create-namespace \
    oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller
```
```
kubectl create secret generic gh-app-auth \
  --namespace=arc-runners \
  --from-literal=github_app_id=<app_id> \
  --from-literal=github_app_installation_id=<installation_id>\
  --from-literal=github_app_private_key='<private_key>'
```
```
helm upgrade --install runners \
    --namespace arc-runners \
    --create-namespace \
    --set githubConfigUrl="https://github.com/<org_name>" \
    --set githubConfigSecret="gh-app-auth" \
    oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set
```

When I go back and look at the logs of the controller, I can see its complaning about  `failed to get runner registration token on refresh: github api error: StatusCode 404` , which leads me to believe it's looking in the wrong place, and if I dig further I can see in the logs its still trying to connect to the original repository url I had setup initially.  Is there some caching mechanism going on?  I've deleted all PVs and PVCs trying to get this to reset but the values are somewhere I can't pinpoint.. any ideas?

**Comment by rredkovich** [comment]
Hi,

Don't know if you managed to fix the issue.
It seems it is not cache error, but something related to Github App. 

I'm having very similar 404 error with the very same set up.

```bash
2025-04-17T15:21:20Z    ERROR   Reconciler error        {"controller": "autoscalingrunnerset", "controllerGroup": "actions.github.com", "controllerKind": "AutoscalingRunnerSet", "AutoscalingRunnerSet": {"name":"arc-runner-set","namespace":"arc-runner"}, "namespace": "arc-runner", "name": "arc-runner-set", "reconcileID": "956b3c47-f940-4e1b-a487-63e15b7fcd77", "error": "failed to get runner registration token on refresh: github api error: StatusCode 404, RequestID \"F951:3B70CD:66308:97C52:68011C6F\": {\"message\":\"Not Found\",\"documentation_url\":\"https://docs.github.com/rest/actions/self-hosted-runners#create-a-registration-token-for-a-repository\",\"status\":\"404\"}"}
```

**Comment by EeshwarChembetiTR** [comment]
You can check the authentication with curl commands
This is a sample I used using pat 

kubectl run github-api-test --image=curlimages/curl -i --rm -- \
  curl -s -H "Authorization: token YOUR_GITHUB_PAT" \
  https://api.github.com/orgs/your-org-name

https://github.com/EeshwarChembetiTR/GitHub-ARC-Usage?tab=readme-ov-file#4-testing-github-api-access
